pkg/operator/vspherecontroller/checks/check_nodes.go

```diff
@@ -242,6 +261,23 @@ func (n *NodeChecker) getHost(ctx context.Context, checkOpts CheckArgs, hostRef
 	return o, nil
 }
 
+func (n *NodeChecker) getMaxVolumeLimit() (int, error) {
+	maxVolumesPerNode := maxVolumesPerNodeVSphere7
+
+	// Get ClusterCSIDriver object
+	clusterCSIDriver, err := n.ClusterCSIDriverLister.Get(utils.VSphereDriverName)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get ClusterCSIDriver: %v", err)
+	}
+
+	// Check if a custom value is set in the ClusterCSIDriver
+	if clusterCSIDriver != nil && clusterCSIDriver.Spec.DriverConfig.VSphere != nil {
+		maxVolumesPerNode = int(clusterCSIDriver.Spec.DriverConfig.VSphere.MaxAllowedBlockVolumesPerNode)
```

Comment:
What does this return when no limit is set? 0 ??

Reply:
Yeah, this needs one more check - added.